osbs/utils.py

```diff
@@ -20,3 +25,25 @@ def graceful_chain_get(d, *args):
         except (AttributeError, KeyError):
             return None
     return t
+
+
+def deep_update(orig, new):
+    if isinstance(orig, dict) and isinstance(new, dict):
+        for k, v in new.items():
+            if isinstance(orig.get(k, None), dict) and isinstance(v, dict):
+                deep_update(orig[k], v)
+            else:
+                orig[k] = v
+
+
+def checkout_git_repo(uri, commit):
+    tmpdir = tempfile.mkdtemp()
+    subprocess.check_call(['git', 'clone', uri, '-b', commit, tmpdir], stdout=subprocess.PIPE,
+                          stderr=subprocess.PIPE)
+    return tmpdir
+
+
+def get_base_image(repo_dir):
+    df_path = os.path.join(repo_dir, 'Dockerfile')
```

Comment:
Currently `dockerfile_parse` requires that the file is named `Dockerfile`. If it isn't, it assumes it is a directory and looks for a `Dockerfile` inside it.